src/cts/src/HTreeBuilder.cpp

```diff
@@ -1209,6 +1209,14 @@ void HTreeBuilder::run()
   // clang-format on
 }
 
+bool HTreeBuilder::isNumberOfSinksTooSmall(unsigned numSinksPerSubRegion) const
```

Comment:
Shouldn't this API return if either of the conditions is met?   < numMaxLeafSinks_  || < options_->getMaxFanout()?